pkg/k8s/watchers/watcher.go

```diff
@@ -610,6 +618,13 @@ func (k *K8sWatcher) k8sServiceHandler() {
 				break
 			} else if result.NumToServicesRules > 0 {
 				// Only trigger policy updates if ToServices rules are in effect
+				_, err := k.ipcache.AllocateCIDRs(result.PrefixesToAdd, nil, nil)
+				if err != nil {
+					scopedLog.WithError(err).
+						Error("Unabled to allocate ipcache CIDR for toService rule")
+					break
+				}
```

Comment:
Is it possible to allocate on a `DeleteService` event? I would have thought that will only remove CIDRs from the policy so we could only do the `ReleaseCIDRIdentitiesByCIDR()` call below.

Reply:
Yup, you're right. If revert is set to true on NewK8sTranslator, then `result.PrefixesToRelease` will never be appended to so all this is a no-op:

https://github.com/cilium/cilium/blob/master/pkg/k8s/rule_translate.go#L50

Removing this section.

Reply:
Also, currently in master, the release would be done before the alllocates. Would this make any difference here.
Regardless, going to make that change as well.

Reply:
In general I would probably be inclined to do an allocate before the release if we think that there may be some overlap between the prefixes, just so that at all times the underlying reference counts for overlapping entries would remain > 0. But in practice I'm not so sure it will make a difference. I'm fine with leaving this in the same order as it is currently.